resources/integration_test/tests/query.yaml

```diff
@@ -188,53 +188,64 @@ tests:
-    expected_precision: [10, 0, 0, 0, 0, 0, 0, 24, 0, 0, 0]
-    expected_scale: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
-    expected_schema_name: ['', '', '', '', '', '', '', '', '', '', '']
+                                 org.bson.BsonValue, org.bson.BsonValue, org.bson.BsonValue, org.bson.BsonValue]
+    expected_column_label: [_id, array, dbPointer, javascript, javascriptWithScope,
+                            maxKey, minKey, object, objectId, regularExpression, symbol, timestamp]
+    expected_column_display_size: [10, 0, 0, 0, 0, 0, 0, 0, 24, 0, 0, 0]
+    expected_precision: [10, 0, 0, 0, 0, 0, 0, 0, 24, 0, 0, 0]
+    expected_scale: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+    expected_schema_name: ['', '', '', '', '', '', '', '', '', '', '', '']
     expected_is_auto_increment: [false, false, false, false, false, false, false,
-                                 false, false, false, false]
-    expected_is_case_sensitive: [false, false, true, true, false, false, false, false,
-                                 true, true, false]
+                                 false, false, false, false, false]
+    expected_is_case_sensitive: [false, false, false, true, true, false, false, false,
+                                 false, true, true, false]
     expected_is_currency: [false, false, false, false, false, false, false, false,
-                           false, false, false]
+                           false, false, false, false]
     expected_is_definitely_writable: [false, false, false, false, false, false, false,
-                                      false, false, false, false]
+                                      false, false, false, false, false]
     expected_is_nullable: [columnNullable, columnNullable, columnNullable, columnNullable,
                            columnNullable, columnNullable, columnNullable, columnNullable, columnNullable,
-                           columnNullable, columnNullable]
+                           columnNullable, columnNullable, columnNullable]
     expected_is_read_only: [true, true, true, true, true, true, true, true, true,
-                            true, true]
+                            true, true, true]
     expected_is_searchable: [true, true, true, true, true, true, true, true, true,
-                             true, true]
+                             true, true, true]
     expected_is_signed: [true, false, false, false, false, false, false, false, false,
-                         false, false]
+                         false, false, false]
     expected_is_writable: [false, false, false, false, false, false, false, false,
-                           false, false, false]
+                           false, false, false, false]
+
+  - description: query_array_and_document_as_string
+    db: integration_test
+    sql: SELECT `array`, `object` FROM types_other
+    expected_result:
+      - ["[{\"$numberInt\": \"1\"}, {\"$numberInt\": \"2\"}, {\"$numberInt\": \"3\"}, \
```

Comment:
small suggestion: let's use single quotes for this string so that we don't need to escape all the double quotes

Reply:
The string written by the JsonWriter in getString() uses double quotes which wouldn't match if I switched these to single quotes in the expected results.

Reply:
It just occurred to me what you meant.  Made the change 👍